File: ruby/reverse_string.rb
def reverse_string(str)
  counter = 0
  while counter < str.length/2 do
    temp = str[counter]
    back_ind = str.length - 1 - counter
    str[counter] = str[back_ind]
    str[back_ind] = temp
    counter += 1
  end
  return str
end

if __FILE__ == $PROGRAM_NAME
  puts "Expecting: 'ih'"
  puts "=>", reverse_string('hi')

  puts

  puts "Expecting: 'ybabtac'"
  puts "=>", reverse_string('catbaby')
  
  puts "Expecting: ''"
  puts "=>", reverse_string('')
end

# Please add your pseudocode to this file
# And a written explanation of your solution

File: solutions/reverse_string.py
def reverse_string(string):

    reversed_string = ""

    for i in range(len(string) - 1, -1, -1):
        reversed_string += string[i]

    return reversed_string

print("Expecting: 'ih'")
print(reverse_string("hi"))

print("")

print("Expecting: 'ybabtac'")
print(reverse_string("catbaby"))

print("")

print("Expecting: 'a'")
print(reverse_string("a"))

print("")

print("Expecting: '' (empty string)")
print(reverse_string(""))